### DISCUSSION TITLE
Computing the right odf from pole figure data obtained from XRD experiment

### DISCUSSION THREAD

**Discussion by Wenhaosu10** [mtex-toolbox/mtex#2080] [Ask Anything] [answered]
Hi guys!

I want to use MTEX to calculate the anisotropy of antigorite and olivine, and the data were collected from the XRD. I have a question about the function of computing odf from the pole figure data. I found that the pole figure results calculated from the pf (figure1) and odf (figure2) are different, and it seems like there might be some problems on the odf variable. So I would like to know how to solve this problem, because I need the right odf to calculate the anisotropy. Thanks!
figure_1 plot by using pf
![pole_figure_1_pf](https://github.com/mtex-toolbox/mtex/assets/161533771/558d15a3-8b06-43a9-912a-a141848083ab)
figure_2 plot by using odf
![pole_figure_2_odf](https://github.com/mtex-toolbox/mtex/assets/161533771/3cd5ec5c-a42f-4474-82e7-4be2297d6b13)
And here are my code and the .XPO files exported from the Beartex.
```
%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = crystalSymmetry('11m', [7 44 8.8], [90,90,90.696]*degree, 'X||a*', 'Y||b', 'Z||c');

% specimen symmetry
SS = specimenSymmetry('1');

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = '/Users/Desktop/Antigorite/nonfiber_mma/maa with no fiber symmetry/D2646_nofiber';

% which files to be imported
fname = [pname '/D2646_120_ant.XPO'];
%% Specify Miller Indice

h = { ...
  Miller(1,0,0,CS),...
  Miller(0,1,0,CS),...
  Miller(0,0,1,CS),...
  };

%% Import the Data

% create a Pole Figure variable containing the data
pf = PoleFigure.load(fname,h,CS,SS,'interface','beartex');

%% Pole Figure
odf_ant = calcODF(pf);
plot(pf,'contourf','minmax','LineWidth',2,'HalfWidth',10*degree);
hold on
annotate(yvector,'label',{'Z'},'BackgroundColor','w')
annotate(xvector,'label',{'X'},'BackgroundColor','w')
mtexColorMap parula
mtexColorbar
hold off
figure ()
plotPDF(odf_ant,Miller({1,0,0},{0,1,0},{0,0,1},CS),'lower','contourf','minmax','LineWidth',2,'HalfWidth',10*degree)
mtexColorMap parula
mtexColorbar
%% seismic anisotropy
CS_Tensor_Ant = crystalSymmetry('11m', [7 44 8.8] , [90,90,90.696]*degree, 'X||a*', 'Y||b', 'Z||c');
rho_Ant = 2.62;
Cij = [[96.7     0     0     0     0   2.5];...
  [  15.9 208.4  66.2     0     0   2.4];...
  [   5     0 201.6     0     0  -4.4];...
  [   0     0     0    65     0     0];...
  [   0     0     0 -13.1  17.4     0];...
  [   0     0     0     0     0  18.3]];
C_Ant = stiffnessTensor(Cij,CS_Tensor_Ant,'density',rho_Ant);
cs_Ant_new = crystalSymmetry('1m1',[44 8.8 7],[90,90.696,90]*degree,'X||a', 'Y||b', 'Z||c*');
Cij_1=[[208.4    66.2   15.9     0     2.4     0];...
  [   0     201.6    5     0     -4.4     0];...
  [   0      0    96.7     0     2.5     0];...
  [   0      0      0     17.4     0     -13.1];...
  [   0      0      0      0    18.3     0];...
  [   0      0      0      0     0      65]];
C_Ant_new = stiffnessTensor(Cij_1,cs_Ant_new,'density',rho_Ant);
odf_ant.CS= crystalSymmetry('1m1',[44 8.8 7],[90,90.696,90]*degree,'X||a', 'Y||b', 'Z||c*');
[CVoigt_ant, CReuss_ant, CHill_ant] = mean(C_Ant_new,odf_ant);
rot = rotation.map(xvector,xvector,yvector,zvector);
rot_CHill_ant = rot * CHill_ant;
figure ()
plotSeismicVelocities(rot_CHill_ant)
```
[D2646_120_ant.zip](https://github.com/mtex-toolbox/mtex/files/14428120/D2646_120_ant.zip)

Best
Wenhao

**Comment by kilir** [comment]
Hi,
I think your symmetry is a little funny. If you do `CS = CS.Laue`  I'd say they look fine.

Cheers,

Rüdiger

**Reply by Wenhaosu10** [reply]
Thanks, Rüdiger!

It works well! And I have another question about the crystal symmetry. The CS convention ([7 44 8.8]) of antigorite in our XRD experiment is different from the traditional convention ([44 8.8 7]), and I want to use the traditional convention to compute the anisotropy. I tried to use the following code to substitute the CS, but the pole figure results are still the same after the substitution. So, I would like to know how to change the CS of the ODF variable.

```
odf_ant = calcODF(pf);
CS_Ant_new = crystalSymmetry('1m1',[44 8.8 7],[90,90.696,90]*degree,'X||a', 'Y||b', 'Z||c*');
odf_ant.CS = CS_Ant_new.Laue;

```
Best,
Wenhao

**Reply by ralfHielscher** [reply, marked as answer]
Hi Wenhao,

in the next release of MTEX we will include a command ```odf_new = odf.transformReferenceFrame(cs_new)``` which will do exactly what you described. It will hopefully released this week. See 62758d64733186037cb4026797042a0415f0236f

Ralf.

**Reply by Wenhaosu10** [reply]
Thanks Ralf!

That would be so great!

Best
Wenhao

**Reply by Wenhaosu10** [reply]
> Hi Wenhao,
> 
> in the next release of MTEX we will include a command `odf_new = odf.transformReferenceFrame(cs_new)` which will do exactly what you described. It will hopefully released this week. See [62758d6](https://github.com/mtex-toolbox/mtex/commit/62758d64733186037cb4026797042a0415f0236f)
> 
> Ralf.

Hi Ralf,

The new function is good and it is very helpful. However, I have a question about the anisotropy calculation by using the ODF file from the XRD experiment. I calculated the anisotropy based on the ODF file and Cij of antigorite, but the value is quite low, which I think is not consistent with the high anisotropy of antigorite single crystal and the relatively strong CPO we got (especially the [001] axes). Therefore, I wonder if it is because the calculation using the ODF file would underestimate the anisotropy as the instruction shown here: https://mtex-toolbox.github.io/CPOSeismicProperties.html, or if there are issues in my code. I attached the code and data I used and the results below. 

Thanks!
Wenhao

![PoleFigure](https://github.com/user-attachments/assets/960260e5-877e-498a-9648-13bc78eb6dca)
![Anisotropy](https://github.com/user-attachments/assets/e67d7d47-1b4a-44fe-bb61-4cb4d047582b)

```
%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = crystalSymmetry('11m', [7 44 8.8], [90,90,90.696]*degree, 'X||a*', 'Y||b', 'Z||c');
CS = CS.Laue;

% specimen symmetry
SS = specimenSymmetry('1');

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'H:\nonfiber_mma\maa with no fiber symmetry\D2608_nofiber';

% which files to be imported
fname = [pname '/D2608_134_ant.XPO'];
%% Specify Miller Indice

h = { ...
  Miller(1,0,0,CS),...
  Miller(0,1,0,CS),...
  Miller(0,0,1,CS),...
  };

%% Import the Data

% create a Pole Figure variable containing the data
pf = PoleFigure.load(fname,h,CS.Laue,SS,'interface','beartex');
rot = rotation.map(xvector,xvector,yvector,zvector);
rot_pf = rot * pf;

%% Pole Figure
odf_ant = calcODF(rot_pf);
cs_new = crystalSymmetry('1m1',[44 8.8 7],[90,90.696,90]*degree,'X||a', 'Y||b', 'Z||c*');
cs_new = cs_new.Laue
odf_ant = transformReferenceFrame(odf_ant,cs_new)
plot(rot_pf,'contourf','minmax','LineWidth',2,'HalfWidth',10*degree);
hold on
annotate(yvector,'label',{'Z'},'BackgroundColor','w')
annotate(xvector,'label',{'X'},'BackgroundColor','w')
mtexColorMap parula
mtexColorbar
hold off
figure ()
plotPDF(odf_ant,Miller({1,0,0},{0,1,0},{0,0,1},cs_new),'lower','contourf','minmax','LineWidth',2,'HalfWidth',10*degree)
mtexColorMap WhiteJet
mtexColorbar
hold on
annotate(yvector,'label',{'Z'},'BackgroundColor','w')
annotate(xvector,'label',{'X'},'BackgroundColor','w')
hold off
%MI = calcMIndex(odf_ant);
%J=textureindex(odf_ant);
%% seismic anisotropy
rho_Ant_5GPa = 2.795;
rho_Ant_3GPa = 2.733;
Cij_3GPa = [[ 180.6   52.8     44    0     5.1   0];...
  [   52.8   206.0     40     0    1     0];...
  [   44     40     153.1     0    0.6     0];...
  [   0     0     0     15    0     -4.3];...
  [   5.1     1     0.6     0    20.3     0];...
  [   0     0     0     -4.3    0     70.5]];
Cij_5GPa = [[186.7    51.9    35.1     0     3.8   0];...
  [  51.9 204.9  31.1     0     1   0];...
  [   35.1  31.1   129.5     0     1.9     0];...
  [  0     0     0  18   0     -2.7];...
  [   3.8     1     1.9  0    19.5     0];...
  [   0   0     0     -2.7     0  70.8]];
C_Ant = stiffnessTensor(Cij_5GPa,cs_new,'density',rho_Ant_5GPa);
[CVoigt_ant, CReuss_ant, CHill_ant] = mean(C_Ant,odf_ant);
vol_ant=1;
[CVoigt, CReuss, CHill] = mean(CVoigt_ant,...
  'weights',vol_ant);
rot = rotation.map(xvector,xvector,yvector,zvector);
rot_CHill_ant = rot * CHill;
figure ()
plotSeismicVelocities(rot_CHill_ant)
```
[D2608_134_ant.zip](https://github.com/user-attachments/files/17581919/D2608_134_ant.zip)
